### DISCUSSION TITLE
Is it reasonable for the degree of resultant curve exceeds the MaxDegree of GeomConvert_ApproxCurve?

### DISCUSSION THREAD

**Discussion by lavender-lee** [Open-Cascade-SAS/OCCT#142] [General]
Consider the following code:

```c++
void test_GeomConvert_ApproxCurve()
{
  double Tol = 0.01;
  GeomAbs_Shape continuity = GeomAbs_C2;
  int max_degree = 3;
  int max_segments = 20;
  //
  gp_Pnt center(0.0, 0.0, 0.0);
  double radius = 10.0;
  gp_Ax2 position(center, gp_Dir(0, 0, 1), gp_Dir(1, 0, 0));
  Handle(Geom_Circle) circle = new Geom_Circle(position, radius);
  GeomConvert_ApproxCurve anApprox3d(circle, Tol, continuity, max_degree, max_segments);
  //
  anApprox3d.Dump(std::cout);
  //
  if (anApprox3d.HasResult())
  {
    const Handle(Geom_BSplineCurve) &curve = anApprox3d.Curve();
    std::cout << "degree: " << curve ->Degree() << std::endl;
  }
}
```

The degree of the approximation curve is **9**, a 9-th degree Bezier curve. 

![image](https://github.com/user-attachments/assets/f2c18b76-e812-4daf-806c-698da8e12560)

However, the document says:

![image](https://github.com/user-attachments/assets/a37808a9-1168-4f22-a109-15154e1adbb0)

In the above case, the setting for **MaxDegree** is only 3.

**Comment by lavender-lee** [comment]
However, approximate a **circle** with a cubic spline within the specified tolerance is possible. For example:

```
poles = {{10.0057, 0.0000149421}, {10.0057, 2.58765}, {7.87306, 7.79652}, {0.0646744, 11.095}, {-7.81643, 7.87408}, {-11.0947, 
  0.017671}, {-7.84136, -7.84914}, {0.0290646, -11.0951}, {7.85936, -7.81712}, {10.0056, -2.59825}, {10.0057, 0.0000149421}}

knots = {0, 0, 0, 0., 0.123743, 0.249077, 0.374411, 0.499746, 0.62508, 0.750414, 0.875749, 1., 1, 1, 1}
```

circle equation in parametric form :

x(t) = 10 Cos[6.28319 t],
y(t) = 10 Sin[6.28319 t]


![image](https://github.com/user-attachments/assets/83c2989c-1945-4e52-9d21-6b6fd809a76a)

Error analysis (two components: `x , y`):

![image](https://github.com/user-attachments/assets/65c8bd30-ec43-4a68-8034-79a300fdfc7f)

**Comment by lavender-lee** [comment]
@dpasukhi Do you think this [throw statement](https://github.com/Open-Cascade-SAS/OCCT/blob/fd4fb824c7ae54cb435205eee14ab7013ebe800e/src/AdvApp2Var/AdvApp2Var_ApproxAFunc2Var.cxx#L212) is redundant?

From the math point of view, `c = max(a, b)`, then`c >= a` and `c >= b`. Hence, `ndu<2*iu+2` is impossible.

```c++
  ndu = Max(myMaxDegInU+1,2*iu+2);
  ndv = Max(myMaxDegInV+1,2*iv+2);
  if (ndu<2*iu+2)
    throw Standard_ConstructionError("AdvApp2Var_ApproxAFunc2Var : UMaxDegree Error");
  if (ndv<2*iv+2)
    throw Standard_ConstructionError("AdvApp2Var_ApproxAFunc2Var : VMaxDegree Error");
```

**Comment by aschoutz** [comment]
Shouldn't you swap max_degree and max_segments in your first example? According to the docs max_degree is the last parameter.

**Comment by lavender-lee** [comment]
@arjan-schouten Thanks:)
